src/develop/pixelpipe_cache.c

```diff
@@ -182,26 +178,29 @@ gboolean dt_dev_pixelpipe_cache_available(dt_dev_pixelpipe_cache_t *cache, const
   return FALSE;
 }
 
-gboolean dt_dev_pixelpipe_cache_get_important(dt_dev_pixelpipe_cache_t *cache, const uint64_t basichash,
-                                         const uint64_t hash, const size_t size,
-                                         void **data, dt_iop_buffer_dsc_t **dsc, char *modname)
-{
-  return dt_dev_pixelpipe_cache_get_weighted(cache, basichash, hash, size, data, dsc, -cache->entries, modname);
-}
-
-gboolean dt_dev_pixelpipe_cache_get(dt_dev_pixelpipe_cache_t *cache, const uint64_t basichash, const uint64_t hash,
-                               const size_t size, void **data, dt_iop_buffer_dsc_t **dsc, char *modname)
+static int _get_oldest_cacheline(dt_dev_pixelpipe_cache_t *cache)
 {
-  return dt_dev_pixelpipe_cache_get_weighted(cache, basichash, hash, size, data, dsc, 0, modname);
+  // we never want the latest used cacheline! It was <= 0 and the weight has increased just now
+  int weight = 1;
+  int id = 0;
+  for(int k = 0; k < cache->entries; k++)
+  {
+    if(cache->used[k] > weight)
+    {
+      weight = cache->used[k];
+      id = k;
+    }
+  }
+  return id;
 }
 
-static int _get_oldest_cacheline(dt_dev_pixelpipe_cache_t *cache)
+static int _get_oldest_used_cacheline(dt_dev_pixelpipe_cache_t *cache)
 {
-  int weight = -1;
+  int weight = 16;
```

Comment:
Also 16 here... Could probably help to use a named number for add a comment.

Reply:
did a comment here explaining why ...